src/main/groovy/io/openliberty/tools/gradle/tasks/DevTask.groovy

```diff
@@ -0,0 +1,503 @@
+        public void checkConfigFile(File configFile, File serverDir) {
+            // TODO:
+        }
+
+        @Override
+        public boolean compile(File dir) {
+            ProjectConnection gradleConnection = initGradleProjectConnection();
+            BuildLauncher gradleBuildLauncher = gradleConnection.newBuild();
+
+            try {
+                if (dir.equals(sourceDirectory)) {
+                    runGradleTask(gradleBuildLauncher, 'compileJava', 'processResources');
+                }
+
+                if (dir.equals(testSourceDirectory)) {
+                    runGradleTask(gradleBuildLauncher, 'compileTestJava', 'processTestResources');
+                }
+                return true;
+            } catch (BuildException e) {
+                logger.error('Unable to compile', e);
+                return false;
+            } finally {
+                gradleConnection.close();
+            }
+        }
+
+        @Override
+        public void runUnitTests() throws PluginExecutionException, PluginScenarioException {
+            // Not needed for gradle.
+        }
+
+        @Override
+        public void runIntegrationTests() throws PluginExecutionException, PluginScenarioException {
+            ProjectConnection gradleConnection = initGradleProjectConnection();
+            BuildLauncher gradleBuildLauncher = gradleConnection.newBuild();
+
+            try {
+                // Force tests to run by calling cleanTest first
+                // otherwise tests may be skipped with an UP-TO-DATE message
+                // https://docs.gradle.org/current/userguide/java_testing.html#sec:forcing_java_tests_to_run
+                runGradleTask(gradleBuildLauncher, 'cleanTest', 'test');
+            } catch (BuildException e) {
+                // Gradle throws a build exception if tests fail
+                // catch it and do nothing
+            } finally {
+                gradleConnection.close();
+            }
+        }
+
+        @Override
+        public void redeployApp() {
+            // TODO:
+        }
+
+        @Override
+        public void libertyInstallFeature() {
+            // TODO: 
+        }
+
+        @Override
+        public void libertyDeploy() {
+            // TODO: 
+        }
+
+        @Override
+        public void libertyCreate() {
+            // TODO: 
+        }
+    }
+
+    @TaskAction
+    void action() {
+            SourceSet mainSourceSet = project.sourceSets.main;
+            SourceSet testSourceSet = project.sourceSets.test;
+
+            sourceDirectory = mainSourceSet.java.srcDirs.iterator().next()
+            testSourceDirectory = testSourceSet.java.srcDirs.iterator().next()
+
+            File outputDirectory = mainSourceSet.java.outputDir;
+            File testOutputDirectory = testSourceSet.java.outputDir;
+            File serverDirectory = getServerDir(project);
+            File configDirectory = new File(project.projectDir, "src/main/liberty/config");
```

Comment:
Can this use `server.configDirectory` instead?

Reply:
It can but I needed to make sure that `server.configDirectory` is set. So I refactored out the logic that sets it in `copyConfigFiles`  in `AbstractServerTask` to an `initializeConfigDirectory` method.